goldmane/pkg/client/client.go

```diff
@@ -52,8 +68,52 @@ func (c *FlowClient) Run(ctx context.Context, grpcClient grpc.ClientConnInterfac
 	// Start the cache cleanup task.
 	go c.cache.Run(FlowCacheCleanup)
 
+	startUp := make(chan struct{})
+	go func() {
+		rc, err := c.connect(ctx)
+		// Close this regardless of the error since we don't want the other side of the channel to hang forever.
+		close(startUp)
+		if err != nil {
+			logrus.WithError(err).Warn("Unable to connect to flow server.")
+			return
```

Comment:
If we hit an error how do we retry? I think we just return and the client never retries?

Reply:
If connect returns an error it's a fatal one (context cancelled for example). So this will block until there's a fatal error or a connection is made (using the back off retry).

In the next loop, if an error occurs using rc then we call connect again, which does the same thing (retry until a connection is established or return if there's a fatal error) then continue processing.

Reply:
I don't understand - there's no lop here, so if we hit this error we `return` but the code that called `Connect()` won't see any indicator of that error.

Reply:
```go
for {
    rc, err := c.connect(ctx)
    do.Once(close(startUp))
    if err != nil {
        ...
}
```

We could put it in the loop (and remove the need for both `connect` calls) if we did something like this

Reply:
> If connect returns an error it's a fatal one (context cancelled for example). So this will block until there's a fatal error or a connection is made (using the back off retry).

Ok.. I see... you're saying we can never recover from any of the errors that connect returns.

I think we need to either exit the process, or hook into liveness / readiness probes

Reply:
Which process do you mean? This does exit the current process (in the client) by returning (the same logic as before).

I think we should hook into some liveness probe or something, but this wasn't what I was trying to iterate on with this PR. I'm not quite trying to change how this works in prod, just the interface in which we interact to make it easier to interact with.

Should I create a ticket to track the liveness probe logic (or process exiting, however this needs to be handled by felix), or did you already have a task for that?

Reply:
Yeah, I suppose it's no worse than it is today.

But goroutine != process here - this error should result in a new gRPC connection attempt, either by restarting Felix altogether or by exiting and creating a new FlowClient altogether (while leaving Felix running).

Reply:
Ya agreed that one of those two things should happen.